src/file.ts

```diff
@@ -3587,7 +3587,7 @@ class File extends ServiceObject<File> {
     if (!callback) {
       return returnValue;
     } else {
-      return returnValue.then(callback);
+      return returnValue.then(callback).catch(callback);
```

Comment:
This looks wrong, but I'm not sure. Should there be two different values for `then()` and `catch()`?

Reply:
The first argument to the closure passed to `catch` will be `err` if it's invoked, the `then` in this case will be invoked with no arguments. So, I believe this behavior works as expected `callback()` when there's a success `callback(err)` on failure.

We should make sure there's a test.

Reply:
@bcoe I believe the error case is already covered by [this system test ](https://github.com/googleapis/nodejs-storage/blob/1d0a7a6aa1c016cc73da2088c083ca571d7aebd9/system-test/storage.ts#L1493) as well as [this unit test](https://github.com/googleapis/nodejs-storage/blob/1d0a7a6aa1c016cc73da2088c083ca571d7aebd9/test/file.ts#L3979). The success case is covered by multiple tests such as [this one ](https://github.com/googleapis/nodejs-storage/blob/1d0a7a6aa1c016cc73da2088c083ca571d7aebd9/test/file.ts#L3944). Is there another method of testing this that you envision?